### ISSUE TITLE
[1.1.1] heapster RC creation failure due to missing NUM_MINIONS variable in minion user data

### ISSUE THREAD

**Issue by romanek-adam** [kubernetes/kubernetes#17285]
Hi, I just started playing with kubernetes v1.1.1 on AWS using the following config:

```
export KUBERNETES_PROVIDER=aws
export KUBE_AWS_ZONE=eu-west-1a
export NUM_MINIONS=1
export MASTER_SIZE=c4.large
export MINION_SIZE=r3.large
export AWS_S3_REGION=eu-west-1
export AWS_S3_BUCKET=xxx
export KUBE_AWS_INSTANCE_PREFIX=xxx
```

Unfortunately the very first thing I encountered was that cluster monitoring addon was broken. Apparently the heapster RC was missing. I found out that the [RC spec for heapster](https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/cluster-monitoring/influxdb/heapster-controller.yaml#L2) relies on `num_nodes` salt variable, which gets defined by [cluster/aws/templates/create-dynamic-salt-files.sh](https://github.com/kubernetes/kubernetes/blob/master/cluster/aws/templates/create-dynamic-salt-files.sh#L43), which on the other hand is invoked by the [k8s master start (user data) script](https://github.com/kubernetes/kubernetes/blob/master/cluster/aws/util.sh#L876). Unfortunately the variable that `create-dynamic-salt-files.sh` relies on (NUM_MINIONS) is not provided to the master start script, hence `num_nodes` gets no value (in fact it evaluates to None, which can be observed later on in salt logs).

I'll provide a patch for this issue but I haven't signed the CLA yet, so feel free to fix this on your own.

BTW, does anyone test Kubernetes on AWS regularly or at least just before a release is made? Is AWS support considered official/stable? I mostly use default configs and this is not the first time I encounter such a fundamental issue.

**Comment by romanek-adam** [comment]
The issue bug exists on master, should I provide a PR too?

**Comment by vishh** [comment]
Yes please.

**Comment by romanek-adam** [comment]
Done.

BTW, it's worth mentioning in v1.1.2 release notes that this issue has been fixed as the first PR made it into v1.1.2.
